File: ljwx-platform-app/src/main/java/com/ljwx/platform/app/domain/entity/TenantBrand.java
package com.ljwx.platform.app.domain.entity;

import com.ljwx.platform.core.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * 租户品牌配置实体
 *
 * @author LJWX Platform
 * @since Phase 38
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class TenantBrand extends BaseEntity {

    /**
     * 主键 ID
     */
    private Long id;

    /**
     * 品牌名称
     */
    private String brandName;

    /**
     * Logo URL
     */
    private String logoUrl;

    /**
     * Favicon URL
     */
    private String faviconUrl;

    /**
     * 主色
     */
    private String primaryColor;

    /**
     * 辅助色
     */
    private String secondaryColor;

    /**
     * 背景色
     */
    private String backgroundColor;

    /**
     * 登录页背景图
     */
    private String loginBgUrl;

    /**
     * 登录页标语
     */
    private String loginSlogan;

    /**
     * 版权信息
     */
    private String copyrightText;

    /**
     * 备案号
     */
    private String icpNumber;

    /**
     * 页脚链接（JSONB）
     */
    private String footerLinks;

    /**
     * 移动端图标
     */
    private String mobileIconUrl;

    /**
     * 移动端启动页
     */
    private String mobileSplashUrl;

    /**
     * 自定义 CSS
     */
    private String customCss;

    /**
     * 页脚链接内部类
     */
    @Data
    public static class FooterLink {
        /**
         * 链接文本
         */
        private String text;

        /**
         * 链接 URL
         */
        private String url;

        /**
         * 是否在新窗口打开
         */
        private Boolean openInNewTab;
    }
}

File: ljwx-platform-admin/src/api/tenantBrand.ts
import request from './request'

/**
 * 页脚链接
 */
export interface FooterLink {
  /** 链接文本 */
  text: string
  /** 链接地址 */
  url: string
}

/**
 * 租户品牌配置 VO
 */
export interface TenantBrandVO {
  /** 主键 */
  id: number
  /** 品牌名称 */
  brandName: string
  /** Logo URL */
  logoUrl: string | null
  /** Favicon URL */
  faviconUrl: string | null
  /** 主色 */
  primaryColor: string
  /** 辅助色 */
  secondaryColor: string | null
  /** 背景色 */
  backgroundColor: string | null
  /** 登录页背景图 */
  loginBgUrl: string | null
  /** 登录页标语 */
  loginSlogan: string | null
  /** 版权信息 */
  copyrightText: string | null
  /** 备案号 */
  icpNumber: string | null
  /** 页脚链接 */
  footerLinks: FooterLink[] | null
  /** 移动端图标 */
  mobileIconUrl: string | null
  /** 移动端启动页 */
  mobileSplashUrl: string | null
  /** 自定义 CSS */
  customCss: string | null
  /** 创建时间 */
  createdTime: string
  /** 更新时间 */
  updatedTime: string
}

/**
 * 租户品牌配置更新 DTO
 */
export interface TenantBrandUpdateDTO {
  /** 品牌名称 */
  brandName: string
  /** Logo URL */
  logoUrl?: string
  /** Favicon URL */
  faviconUrl?: string
  /** 主色 */
  primaryColor?: string
  /** 辅助色 */
  secondaryColor?: string
  /** 背景色 */
  backgroundColor?: string
  /** 登录页背景图 */
  loginBgUrl?: string
  /** 登录页标语 */
  loginSlogan?: string
  /** 版权信息 */
  copyrightText?: string
  /** 备案号 */
  icpNumber?: string
  /** 页脚链接 */
  footerLinks?: FooterLink[]
  /** 移动端图标 */
  mobileIconUrl?: string
  /** 移动端启动页 */
  mobileSplashUrl?: string
  /** 自定义 CSS */
  customCss?: string
}

/**
 * 获取当前租户品牌配置
 */
export function getTenantBrand(): Promise<TenantBrandVO> {
  return request.get<TenantBrandVO>('/api/v1/tenant/brand')
}

/**
 * 更新租户品牌配置
 */
export function updateTenantBrand(data: TenantBrandUpdateDTO): Promise<void> {
  return request.put<void>('/api/v1/tenant/brand', data)
}
